source/agent/analytics/videoGstPipeline/VideoGstAnalyzer.cc

```diff
@@ -98,7 +97,7 @@ void VideoGstAnalyzer::clearPipeline()
  
     }
 
-int VideoGstAnalyzer::createPipeline() {
+bool VideoGstAnalyzer::createPipeline() {
 
     loop = g_main_loop_new(NULL, FALSE);
```

Comment:
Please create the g_main_loop after pipeline_ is successfully created. You unref conditionally in clearPlugin, so there might be a memory leak here.